### PR TITLE
expression: enforce error check

### PR DESCRIPTION
**Pull request by zz-jason**
to #4668

### CODE REVIEW

**Comment by shenli** [comment]
LGTM

File: expression/constant_propagation.go [hunk 1]
```diff
@@ -118,7 +118,7 @@ func (s *propagateConstantSolver) propagateEQ() {
 	visited := make([]bool, len(s.conditions))
 	for i := 0; i < MaxPropagateColsCnt; i++ {
 		mapper := s.pickNewEQConds(visited)
-		if mapper == nil || len(mapper) == 0 {
```
Comment (XuHuaiyu) [inline expression/constant_propagation.go]: mapper will not be nil?

Comment (zz-jason) [reply expression/constant_propagation.go]: len(nil) == 0

Comment (XuHuaiyu) [reply expression/constant_propagation.go]: ok

**Pull request merged**